src/renderer/modals/Receive/steps/StepAccount.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+import React, { useCallback, useMemo } from "react";
+import { Trans } from "react-i18next";
+import type {
+  Account,
+  AccountLike,
+  TokenCurrency,
+  CryptoCurrency,
+} from "@ledgerhq/live-common/lib/types";
+import { getMainAccount, getReceiveFlowError } from "@ledgerhq/live-common/lib/account";
+import { listTokensForCryptoCurrency } from "@ledgerhq/live-common/lib/currencies";
+import TrackPage from "~/renderer/analytics/TrackPage";
+import Box from "~/renderer/components/Box";
+import Label from "~/renderer/components/Label";
+import Button from "~/renderer/components/Button";
+import SelectAccount from "~/renderer/components/SelectAccount";
+import SelectCurrency from "~/renderer/components/SelectCurrency";
+import CurrencyDownStatusAlert from "~/renderer/components/CurrencyDownStatusAlert";
+import ErrorBanner from "~/renderer/components/ErrorBanner";
+import type { StepProps } from "../Body";
+
+type OnChangeAccount = (account: ?AccountLike, tokenAccount: ?Account) => void;
+
+const AccountSelection = ({
+  onChangeAccount,
+  account,
+}: {
+  onChangeAccount: OnChangeAccount,
+  account: ?AccountLike,
+}) => (
+  <>
+    <Label>
+      <Trans i18nKey="receive.steps.chooseAccount.label" />
+    </Label>
+    <SelectAccount autoFocus withSubAccounts onChange={onChangeAccount} value={account} />
+  </>
+);
+
+const TokenParentSelection = ({
+  onChangeAccount,
+  mainAccount,
+}: {
+  onChangeAccount: OnChangeAccount,
+  mainAccount: Account,
+}) => {
+  const filterAccountSelect = useCallback(account => account.currency === mainAccount.currency, [
+    mainAccount,
+  ]);
+  return (
+    <>
+      <Label>
+        <Trans
+          i18nKey="receive.steps.chooseAccount.parentAccount"
+          values={{
+            currencyName: mainAccount.currency.name,
+          }}
+        />
+      </Label>
+      <SelectAccount filter={filterAccountSelect} onChange={onChangeAccount} value={mainAccount} />
+    </>
+  );
+};
+
+const TokenSelection = ({
+  currency,
+  token,
+  onChangeToken,
+}: {
+  currency: CryptoCurrency,
+  token: ?TokenCurrency,
+  onChangeToken: (token: ?TokenCurrency) => void,
+}) => {
+  const tokens = useMemo(() => listTokensForCryptoCurrency(currency), [currency]);
+  return (
+    <>
+      <Label mt={30}>
+        <Trans i18nKey="receive.steps.chooseAccount.token" />
+      </Label>
+      {/* I just don't know how to please you anymore Flow */}
+      {/* $FlowFixMe */}
```

Comment:
try to set `currencies: Currency[] | TokenCurrency[] | CryptoCurrency[]` in SelectCurrency prop

meanwhile we'll merge

Reply:
That only brings more errors
![](https://media3.giphy.com/media/9Y5BbDSkSTiY8/giphy.gif)